lib/twilioconnection.js

```diff
@@ -197,6 +195,9 @@ class TwilioConnection extends StateMachine {
         this.emit('error', error);
         return;
       }
+
+      // Each incoming message should be treated as an incoming heartbeat
+      this._handleHeartbeat();
```

Comment:
From our discussion on slack I don't think you should treat each sent or received message as a heartbeat. I was planning on rolling back this change in C++ since it is out of sync with the Server and we don't have agreement between teams. The concern I have is that treating every message as a heartbeat allows the connection to become further and further behind.

Reply:
@ceaglest I understand that we shouldn't treat outgoing message as heartbeat - because server expects heartbeats. But don't see why not for incoming messages. If we miss heartbeat but received incoming message why should we disconnect ?
Can you elaborate more on 
>  treating every message as a heartbeat allows the connection to become further and further behind.

Reply:
resolved after the offline discussion - we decided to treat every incoming message as incoming heartbeat, and outgoing message as outgoing heartbeat. since server already [does the same](https://code.hq.twilio.com/twilio/twilio-connections/blob/master/connections-core/src/main/java/com/twilio/connections/actor/ServerConnectionActor.java#L559) -

Reply:
resolved after offline discussion - we decided to treat incoming message as heartbeat.